docs/contributors.md

```diff
@@ -0,0 +1,123 @@
+- **Be respectful** - follow the [code of conduct](https://opensource.guide/code-of-conduct/)
+
+## Fork Setup for Image Builds
+
+When you fork this repository and push changes, GitHub Actions will automatically:
+- Build multi-architecture container images (amd64 and arm64)
+- Push images to your GitHub Container Registry at `ghcr.io/<your-username>/container-image-csi-driver:<branch>`
+- Make images available for testing your changes before creating a PR
+
+### Setup Steps
+
+#### 1. Enable GitHub Actions
+
+1. Go to your fork: `https://github.com/<your-username>/container-image-csi-driver`
+2. Click **Settings** → **Actions** → **General**
+3. Under **"Actions permissions"**, select: **"Allow all actions and reusable workflows"**
+4. Click **Save**
+
+#### 2. Configure Workflow Permissions
+
+In the same **Settings** → **Actions** → **General** page:
+
+1. Scroll down to **"Workflow permissions"**
+2. Select: **"Read and write permissions"**
+3. Click **Save**
+
+This allows GitHub Actions to push images to your GitHub Container Registry (GHCR).
+
+#### 3. Push and Build
+
+Once configured, simply push to any branch:
+
+```bash
+git push origin <your-branch>
+```
+
+GitHub Actions will automatically build and push to: `ghcr.io/<your-username>/container-image-csi-driver:<your-branch>`
+
+### Using Your Fork Images for Testing
+
+```bash
+# Pull your fork's image
+docker pull ghcr.io/<your-username>/container-image-csi-driver:<branch>
+
+# Or use in Helm
+helm install my-csi charts/warm-metal-csi-driver \
+  --set csiPlugin.image.repository=ghcr.io/<your-username>/container-image-csi-driver \
+  --set csiPlugin.image.tag=<branch>
+```
+
+## Troubleshooting
+
+### Permission denied while pushing to GHCR
+
+**Cause:** Workflow permissions not enabled or PAT expired.
+
+**Solution:**
+1. Verify **Step 2** above (Read and write permissions)
+2. If using a Personal Access Token (PAT), check if it's expired:
+   - Go to GitHub → **Settings** → **Developer settings** → **Personal access tokens**
+   - Create new token with `write:packages` scope
+   - Add to your fork: **Settings** → **Secrets** → Name: `GHCR_TOKEN`
+
+### Actions cannot push to package after first successful build
+
+**Cause:** After the first package is created, GitHub Actions may lose write access to the package.
+
+**Solution:**
+1. Go to your package settings: `https://github.com/users/<your-username>/packages/container/container-image-csi-driver/settings`
+2. Scroll to **"Manage Actions access"** section
+3. Click **"Add Repository"**
+4. Select your forked repository: `<your-username>/container-image-csi-driver`
+5. Set the role to **"Write"** or **"Admin"**
+6. Click **"Add repository"**
+
+This grants your fork's GitHub Actions workflow permission to push subsequent image builds.
+
+### Build is very slow (>20 minutes)
+
+**Expected:** ARM64 builds use QEMU emulation on AMD64 runners.
+- First build: ~25 minutes (no cache)
+- Subsequent builds: ~5-15 minutes (with cache)
```

Comment:
Late to the party, but check out https://github.blog/changelog/2025-01-16-linux-arm64-hosted-runners-now-available-for-free-in-public-repositories-public-preview/, meaning you probably can get away without using QEMU and using native builds